include/rsl/no_discard.hpp

```diff
@@ -12,9 +12,11 @@ namespace rsl {
  * @tparam Fn  Lambda
  */
 template <typename Fn>
-struct NoDiscard {
+class NoDiscard {
     static_assert(std::is_invocable_v<Fn()>, "Fn must be invocable");
     Fn fn_;
+
+   public:
     NoDiscard(Fn const& fn) : fn_(fn) {}
     template <typename... Ts>
     [[nodiscard]] constexpr auto operator()(Ts&&... args) const
```

Comment:
Why would clang tidy want you to make this change? Do you have a link to the rule for this?

Reply:
The style of `fn_` with the trailing underscore indicates that this should be a private member but it was public. That made me realize that we don't actually need to expose that member and should instead make it private unless we later decide to make it public and write tests accordingly.

Reply:
Reading your comment, I think I understand it a bit better. I guess I'm ok with this, although I'm moving towards opposing using private state at all.